### DISCUSSION TITLE
Java Project isn't never load with codespaces

### DISCUSSION THREAD

**Discussion by messeant** [community/community#49863] [Codespaces] [answered]
### Select Topic Area

Question

### Body

Hello

I have try "vscode-remote-try-java" with codespaces.

But Java Project isn't never load
![image](https://user-images.githubusercontent.com/6958444/224561433-26ac97cf-ab78-40cd-85dd-41f549681ce9.png)

I can't configure Java Runtime 
![image](https://user-images.githubusercontent.com/6958444/224561554-53f3eb9a-cee4-4201-88e1-b9c933d4e6b1.png)

my devcontainer (like the turorial)
![image](https://user-images.githubusercontent.com/6958444/224561625-17f7aab8-41db-4b21-ba3b-a5329260435e.png)

Can you help me?

**Comment by bkpecho** [comment]
Hey @messeant 

Check out this [guide](https://docs.github.com/en/codespaces/setting-up-your-project-for-codespaces/adding-a-dev-container-configuration/setting-up-your-java-project-for-codespaces) on setting up a Java project for Codespaces on GitHub. It should help with any issues you're facing.

Let me know if you need any further help.

**Reply by messeant** [reply]
Hello,

I tried codespaces from this tutorial. I don't see why this isn't working.

**Reply by bkpecho** [reply]
Please watch this [video](https://youtu.be/NpIVW4DT434) to learn about essential GitHub usage and how to create a GitHub codespace for Java development in this course. I hope it helps you on your coding journey.

Good luck and have a great day!

**Comment by messeant** [comment]
> Please watch this [video](https://youtu.be/NpIVW4DT434) to learn about essential GitHub usage and how to create a GitHub codespace for Java development in this course. I hope it helps you on your coding journey.
> 
> Good luck and have a great day!

Hello. 
This video does not help me solve my problem.
This does not explain how to configure the version of java.
In this video, he doesn't even use the java project explorer.

**Comment by jungaretti** [comment]
This seems like a problem with the Java extension for VS Code. I'd suggest you look at open issues (or create a new issue) for the repo that you're working with: https://github.com/microsoft/vscode-remote-try-java/issues

It's definitely possible that another GitHub user might have run into this same issue and can help, but the GitHub Community on Discussions focuses primarily on topics related to GitHub itself or collaboration on project development and ideas. We want to make sure you’re getting the best support you can, but this space may not be the right place for this particular topic.

**Comment by rquinteroTecSin** [comment]
When I have a similar problem like this (Java Project not load) I do the following:
1.- I've never use the devcontainer.json which is automatically generated in vs-code. Instead I use a more simplified devcontainer.json including only the [vscode java extensions](https://gist.github.com/ricardo-qm/9850cd1831f06b97afdce23d842d2602). Not more, not less. With this devcontainer.json, codespaces is generated in its image default which functions well (and better) with Java (at least for me).
2.-To load the Java Project with codespace in the navigator some times it not load (which is maybe the case  you have). In this case I load manually the project using the status bar. Pass your cursor over the "Java{}" message in the status bar and a little message appear which the "Load option" available for this. Then, clic over this option and the Java project is loaded without problem.
3.-I have observed that if I load my Java Project in codespaces invoking the local VS Code client (is not necessary that Java extensions are installed locally, local VS Code client use the remote Java extension), the Java Project load without problem and more easy.

I hope this resolve your problem.

**Comment by messeant** [comment, marked as answer]
Thank you all.

I found in from the link of **JP Ungarettia** solution.

I don't understand why it works but it works for "vscode-remote-try-java":

Put in the parameters: 
`"java.server.launchMode": "Standard"`

if you understand that I'm interested in understanding :-)

Thanks again.
